lib/type-annotations.ts

```diff
@@ -0,0 +1,106 @@
+import { Fork } from "../types";
+import typesPlugin, { Field, Type } from "./types";
+import { assertNever } from "./utils";
+
+const Op = Object.prototype;
+const hasOwn = Op.hasOwnProperty;
+
+export interface TSTypeAnnotatorConfig {
+  getReferenceToKind(typeName: string): any;
+}
+
+export default function typeAnnotationsPlugin(fork: Fork) {
+  const { builders, namedTypes } = fork.use(typesPlugin);
+
+  function createTSTypeAnnotator(config: TSTypeAnnotatorConfig) {
+    function getTSTypeAnnotation(type: Type<any>): any {
+      const privateType = type.__type;
```

Comment:
How are you feeling about this approach? It did feel weird to expose this private field in this way. But, I liked how it allowed us to move the type annotation logic into a separate isolated plugin.

Alternatively, we could rename `PrivateType` to `Type` and use that as the common `type` type and expose functions that take in that type like assert/check/getTSTypeAnnotation etc. like `function check<T>(type: Type<T>, value: any, deep?: Deep): value is T { ... }`. Would make things more object-centric and functional vs classes.

Reply:
Made a pass at unifying the Type class and tagged union Type approach here https://github.com/benjamn/ast-types/pull/300/commits/20a5ae2d0085a8966e5addbd30ba013053c938b3

How does that feel to you?